receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -139,20 +141,27 @@ func (mg *metricGroupPdata) toDistributionPoint(orderedLabelKeys []string, dest
 		if i != 0 {
 			adjustedCount -= mg.complexValue[i-1].value
 		}
+		// Buckets still need to be sent to know to set them as stale,
+		// but a staleness NaN converted to uint64 would be an extremely large number.
+		// Setting to 0 instead.
+		if pointIsStale {
+			adjustedCount = 0
+		}
```

Comment:
```suggestion
		// Buckets still need to be sent to know to set them as stale,
		// but a staleness NaN converted to uint64 would be an extremely large number.
		// Setting to 0 instead.
		if pointIsStale {
			adjustedCount = 0
		} else if i != 0 {
			adjustedCount -= mg.complexValue[i-1].value
		}
```

No point adjusting the count twice if the data point is stale.